[WebCore] Cache inline styles

Constellation opened the pull request:
#### a8cb9e6d85cf4333473241f52f4de68b7ae352bf
<pre>
[WebCore] Cache inline styles
<a href="https://bugs.webkit.org/show_bug.cgi?id=270521">https://bugs.webkit.org/show_bug.cgi?id=270521</a>
<a href="https://rdar.apple.com/124074797">rdar://124074797</a>

Reviewed by Antti Koivisto.

CSS in JS is frequently used in the wild. As a result, we can see many
duplicate inline styles randomly, even outside of shadow DOM. Sometimes,
even we use very large strings for inline styles. So we should cache the
inline styles even if it is not inside shadow DOM. This patch adds following
two changes.

1. Caching inline styles for non shadow DOM elements.
2. Adding the mechanism to scan base URL dependecy in CSS rules so that we can disable caching for that case.
   Right now, it is a bit conservative. We disable caching when CSS custom properties exist for example.
3. Originally, we considered using the same mechanism to `setReplacementURLForSubresources` since it needs to
   scan CSSValues too. But it turned out that the implementation is broken and we cannot use it. Since that
   feature also needs to have the correct mechanism to scan CSSValues anyway, this patch implements customVisitChildren
   mechanism to CSSValue, which lists up CSSValue children. So by using this feature, we implement mayDependOnBaseURL
   function by scanning CSSValues. Existing buggy `setReplacementURLForSubresources` / `clearReplacementURLForSubresources` / `traverseSubresources`
   can be rewritten by using this `visitChildren` function. But this patch&apos;s intent is not fixing these existing bugs, so
   we separate the fix from this change. (<a href="https://bugs.webkit.org/show_bug.cgi?id=270600)">https://bugs.webkit.org/show_bug.cgi?id=270600)</a>
4. We found a bug in StyleSheetContents where we didn&apos;t check m_namespaceRules size for `isCacheable()` condition (As a result,
   we observed crashes in some of LayoutTests). This patch fixes it so that we do not say `isCacheable()` =&gt; true when there is
   m_namespaceRules.

* Source/WebCore/css/CSSBasicShapes.h:
(WebCore::CSSInsetShapeValue::top const): Deleted.
(WebCore::CSSInsetShapeValue::right const): Deleted.
(WebCore::CSSInsetShapeValue::bottom const): Deleted.
(WebCore::CSSInsetShapeValue::left const): Deleted.
(WebCore::CSSInsetShapeValue::protectedTop const): Deleted.
(WebCore::CSSInsetShapeValue::protectedRight const): Deleted.
(WebCore::CSSInsetShapeValue::protectedBottom const): Deleted.
(WebCore::CSSInsetShapeValue::protectedLeft const): Deleted.
(WebCore::CSSInsetShapeValue::topLeftRadius const): Deleted.
(WebCore::CSSInsetShapeValue::topRightRadius const): Deleted.
(WebCore::CSSInsetShapeValue::bottomRightRadius const): Deleted.
(WebCore::CSSInsetShapeValue::bottomLeftRadius const): Deleted.
(WebCore::CSSInsetShapeValue::protectedTopLeftRadius const): Deleted.
(WebCore::CSSInsetShapeValue::protectedTopRightRadius const): Deleted.
(WebCore::CSSInsetShapeValue::protectedBottomRightRadius const): Deleted.
(WebCore::CSSInsetShapeValue::protectedBottomLeftRadius const): Deleted.
(WebCore::CSSCircleValue::radius const): Deleted.
(WebCore::CSSCircleValue::centerX const): Deleted.
(WebCore::CSSCircleValue::centerY const): Deleted.
(WebCore::CSSCircleValue::protectedRadius const): Deleted.
(WebCore::CSSCircleValue::protectedCenterX const): Deleted.
(WebCore::CSSCircleValue::protectedCenterY const): Deleted.
(WebCore::CSSEllipseValue::radiusX const): Deleted.
(WebCore::CSSEllipseValue::radiusY const): Deleted.
(WebCore::CSSEllipseValue::centerX const): Deleted.
(WebCore::CSSEllipseValue::centerY const): Deleted.
(WebCore::CSSEllipseValue::protectedRadiusX const): Deleted.
(WebCore::CSSEllipseValue::protectedRadiusY const): Deleted.
(WebCore::CSSEllipseValue::protectedCenterX const): Deleted.
(WebCore::CSSEllipseValue::protectedCenterY const): Deleted.
(WebCore::CSSPolygonValue::windRule const): Deleted.
(WebCore::CSSRectShapeValue::top const): Deleted.
(WebCore::CSSRectShapeValue::right const): Deleted.
(WebCore::CSSRectShapeValue::bottom const): Deleted.
(WebCore::CSSRectShapeValue::left const): Deleted.
(WebCore::CSSRectShapeValue::protectedTop const): Deleted.
(WebCore::CSSRectShapeValue::protectedRight const): Deleted.
(WebCore::CSSRectShapeValue::protectedBottom const): Deleted.
(WebCore::CSSRectShapeValue::protectedLeft const): Deleted.
(WebCore::CSSRectShapeValue::topLeftRadius const): Deleted.
(WebCore::CSSRectShapeValue::topRightRadius const): Deleted.
(WebCore::CSSRectShapeValue::bottomRightRadius const): Deleted.
(WebCore::CSSRectShapeValue::bottomLeftRadius const): Deleted.
(WebCore::CSSRectShapeValue::protectedTopLeftRadius const): Deleted.
(WebCore::CSSRectShapeValue::protectedTopRightRadius const): Deleted.
(WebCore::CSSRectShapeValue::protectedBottomRightRadius const): Deleted.
(WebCore::CSSRectShapeValue::protectedBottomLeftRadius const): Deleted.
(WebCore::CSSXywhValue::insetX const): Deleted.
(WebCore::CSSXywhValue::insetY const): Deleted.
(WebCore::CSSXywhValue::width const): Deleted.
(WebCore::CSSXywhValue::height const): Deleted.
(WebCore::CSSXywhValue::protectedInsetX const): Deleted.
(WebCore::CSSXywhValue::protectedInsetY const): Deleted.
(WebCore::CSSXywhValue::protectedWidth const): Deleted.
(WebCore::CSSXywhValue::protectedHeight const): Deleted.
(WebCore::CSSXywhValue::topLeftRadius const): Deleted.
(WebCore::CSSXywhValue::topRightRadius const): Deleted.
(WebCore::CSSXywhValue::bottomRightRadius const): Deleted.
(WebCore::CSSXywhValue::bottomLeftRadius const): Deleted.
(WebCore::CSSXywhValue::protectedTopLeftRadius const): Deleted.
(WebCore::CSSXywhValue::protectedTopRightRadius const): Deleted.
(WebCore::CSSXywhValue::protectedBottomRightRadius const): Deleted.
(WebCore::CSSXywhValue::protectedBottomLeftRadius const): Deleted.
(WebCore::CSSPathValue::pathData const): Deleted.
(WebCore::CSSPathValue::windRule const): Deleted.
* Source/WebCore/css/CSSCounterValue.h:
(WebCore::CSSCounterValue::identifier const): Deleted.
(WebCore::CSSCounterValue::separator const): Deleted.
(WebCore::CSSCounterValue::counterStyle const): Deleted.
* Source/WebCore/css/CSSCrossfadeValue.h:
* Source/WebCore/css/CSSCursorImageValue.h:
* Source/WebCore/css/CSSCustomPropertyValue.h:
* Source/WebCore/css/CSSFilterImageValue.h:
* Source/WebCore/css/CSSFontFaceSrcValue.cpp:
(WebCore::CSSFontFaceSrcResourceValue::customMayDependOnBaseURL const):
* Source/WebCore/css/CSSFontFaceSrcValue.h:
* Source/WebCore/css/CSSFontPaletteValuesOverrideColorsValue.h:
* Source/WebCore/css/CSSFontStyleRangeValue.h:
* Source/WebCore/css/CSSFontStyleWithAngleValue.h:
* Source/WebCore/css/CSSFontValue.cpp:
(WebCore::CSSFontValue::customVisitChildren const):
* Source/WebCore/css/CSSFontValue.h:
* Source/WebCore/css/CSSGradientValue.h:
* Source/WebCore/css/CSSImageSetOptionValue.h:
* Source/WebCore/css/CSSImageValue.cpp:
(WebCore::CSSImageValue::customMayDependOnBaseURL const):
* Source/WebCore/css/CSSImageValue.h:
* Source/WebCore/css/CSSOffsetRotateValue.h:
* Source/WebCore/css/CSSPendingSubstitutionValue.h:
(WebCore::CSSPendingSubstitutionValue::customVisitChildren const):
* Source/WebCore/css/CSSPrimitiveValue.cpp:
(WebCore::CSSPrimitiveValue::customVisitChildren const):
* Source/WebCore/css/CSSPrimitiveValue.h:
* Source/WebCore/css/CSSRayValue.h:
* Source/WebCore/css/CSSReflectValue.h:
* Source/WebCore/css/CSSScrollValue.h:
* Source/WebCore/css/CSSShadowValue.h:
* Source/WebCore/css/CSSValue.cpp:
(WebCore::CSSValue::visitChildren const):
(WebCore::CSSValue::mayDependOnBaseURL const):
* Source/WebCore/css/CSSValue.h:
(WebCore::CSSValue::customMayDependOnBaseURL const):
(WebCore::CSSValue::customVisitChildren const):
* Source/WebCore/css/CSSValueList.h:
(WebCore::CSSValueContainingVector::customVisitChildren const):
* Source/WebCore/css/CSSValuePair.h:
(WebCore::CSSValuePair::customVisitChildren const):
* Source/WebCore/css/CSSVariableReferenceValue.h:
(WebCore::CSSVariableReferenceValue::customMayDependOnBaseURL const):
* Source/WebCore/css/CSSViewValue.h:
* Source/WebCore/css/StyleProperties.cpp:
(WebCore::StyleProperties::mayDependOnBaseURL const):
* Source/WebCore/css/StyleProperties.h:
* Source/WebCore/css/StyleSheetContents.cpp:
(WebCore::StyleSheetContents::StyleSheetContents):
(WebCore::StyleSheetContents::isCacheable const):
(WebCore::StyleSheetContents::isCacheableAndNoBaseURLDependency const):
(WebCore::StyleSheetContents::mayDependOnBaseURL const):
* Source/WebCore/css/StyleSheetContents.h:
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::mayDependOnBaseURL):
* Source/WebCore/css/parser/CSSParserContext.h:
* Source/WebCore/dom/InlineStyleSheetOwner.cpp:
(WebCore::InlineStyleSheetOwner::createSheet):
(WebCore::makeStyleSheetContentsCacheKey): Deleted.

Canonical link: <a href="https://commits.webkit.org/275795@main">https://commits.webkit.org/275795@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a5b8a0dc5fa11afb086e1e1ce681e54ee8603a82

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42867 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21888 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45268 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45485 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38992 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45173 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25586 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19268 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/35474 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43440 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18921 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/36910 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16446 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16531 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/37969 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/923 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39052 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/38303 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47006 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17694 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14589 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/42223 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19319 "Built successfully") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/40872 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19483 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5802 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18949 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

anttijk commented on Source/WebCore/css/CSSCustomPropertyValue.h:
```diff
@@ -107,6 +107,18 @@ class CSSCustomPropertyValue final : public CSSValue {
 
     Ref<const CSSVariableData> asVariableData() const;
 
+    // Say true conservatively.
+    bool customMayDependOnBaseURL() const { return true; }
```
Did you actually see problems with allowing this? Custom properties are resolved and applied dynamically during style resolution so they shouldn't affect cacheability.

Constellation replied:
One of the issue I found is that `CSSCustomPropertySyntax::Type::URL` can include URL which can be a resolved with a BaseURL. And to detect this, we need to have some SyntaxValue traversing code too. So for now, I disabled this.

anttijk commented on Source/WebCore/css/CSSVariableReferenceValue.h:
```diff
@@ -68,6 +68,9 @@ class CSSVariableReferenceValue : public CSSValue {
     template<typename CacheFunction> bool resolveAndCacheValue(Style::BuilderState&, CacheFunction&&) const;
 
+    // Say true conservatively.
+    bool customMayDependOnBaseURL() const { return true; }
```
Same question here.

Constellation replied:
On the other hand, I think this is not necessary to prohibit it. I'll drop this.

anttijk commented on Source/WebCore/css/StyleSheetContents.h:
```diff
@@ -72,6 +72,7 @@ class StyleSheetContents final : public RefCounted<StyleSheetContents>, public C
     WEBCORE_EXPORT bool parseString(const String&);
 
     bool isCacheable() const;
+    bool isCacheableAndNoBaseURLDependency() const;
```
`isCacheableWithNoBaseURLDependency` might read bit better.

Constellation replied:
Changed.

anttijk commented on Source/WebCore/dom/InlineStyleSheetOwner.cpp:
```diff
@@ -51,16 +51,6 @@ static CSSParserContext parserContextForElement(const Element& element)
-static std::optional<Style::StyleSheetContentsCache::Key> makeStyleSheetContentsCacheKey(const String& text, const Element& element)
-{
-    // Only cache for shadow trees. Main document inline stylesheets are generally unique and can't be shared between documents.
-    // FIXME: This could be relaxed when a stylesheet does not contain document-relative URLs (or #urls).
```
Nice!

Constellation replied:
:D

Constellation committed:
[WebCore] Cache inline styles
https://bugs.webkit.org/show_bug.cgi?id=270521
rdar://124074797

Reviewed by Antti Koivisto.

CSS in JS is frequently used in the wild. As a result, we can see many
duplicate inline styles randomly, even outside of shadow DOM. Sometimes,
even we use very large strings for inline styles. So we should cache the
inline styles even if it is not inside shadow DOM. This patch adds following
two changes.

1. Caching inline styles for non shadow DOM elements.
2. Adding the mechanism to scan base URL dependecy in CSS rules so that we can disable caching for that case.
   Right now, it is a bit conservative. We disable caching when CSS custom properties exist for example.
3. Originally, we considered using the same mechanism to `setReplacementURLForSubresources` since it needs to
   scan CSSValues too. But it turned out that the implementation is broken and we cannot use it. Since that
   feature also needs to have the correct mechanism to scan CSSValues anyway, this patch implements customVisitChildren
   mechanism to CSSValue, which lists up CSSValue children. So by using this feature, we implement mayDependOnBaseURL
   function by scanning CSSValues. Existing buggy `setReplacementURLForSubresources` / `clearReplacementURLForSubresources` / `traverseSubresources`
   can be rewritten by using this `visitChildren` function. But this patch's intent is not fixing these existing bugs, so
   we separate the fix from this change. (https://bugs.webkit.org/show_bug.cgi?id=270600)
4. We found a bug in StyleSheetContents where we didn't check m_namespaceRules size for `isCacheable()` condition (As a result,
   we observed crashes in some of LayoutTests). This patch fixes it so that we do not say `isCacheable()` => true when there is
   m_namespaceRules.

* Source/WebCore/css/CSSBasicShapes.h:
(WebCore::CSSInsetShapeValue::top const): Deleted.
(WebCore::CSSInsetShapeValue::right const): Deleted.
(WebCore::CSSInsetShapeValue::bottom const): Deleted.
(WebCore::CSSInsetShapeValue::left const): Deleted.
(WebCore::CSSInsetShapeValue::protectedTop const): Deleted.
(WebCore::CSSInsetShapeValue::protectedRight const): Deleted.
(WebCore::CSSInsetShapeValue::protectedBottom const): Deleted.
(WebCore::CSSInsetShapeValue::protectedLeft const): Deleted.
(WebCore::CSSInsetShapeValue::topLeftRadius const): Deleted.
(WebCore::CSSInsetShapeValue::topRightRadius const): Deleted.
(WebCore::CSSInsetShapeValue::bottomRightRadius const): Deleted.
(WebCore::CSSInsetShapeValue::bottomLeftRadius const): Deleted.
(WebCore::CSSInsetShapeValue::protectedTopLeftRadius const): Deleted.
(WebCore::CSSInsetShapeValue::protectedTopRightRadius const): Deleted.
(WebCore::CSSInsetShapeValue::protectedBottomRightRadius const): Deleted.
(WebCore::CSSInsetShapeValue::protectedBottomLeftRadius const): Deleted.
(WebCore::CSSCircleValue::radius const): Deleted.
(WebCore::CSSCircleValue::centerX const): Deleted.
(WebCore::CSSCircleValue::centerY const): Deleted.
(WebCore::CSSCircleValue::protectedRadius const): Deleted.
(WebCore::CSSCircleValue::protectedCenterX const): Deleted.
(WebCore::CSSCircleValue::protectedCenterY const): Deleted.
(WebCore::CSSEllipseValue::radiusX const): Deleted.
(WebCore::CSSEllipseValue::radiusY const): Deleted.
(WebCore::CSSEllipseValue::centerX const): Deleted.
(WebCore::CSSEllipseValue::centerY const): Deleted.
(WebCore::CSSEllipseValue::protectedRadiusX const): Deleted.
(WebCore::CSSEllipseValue::protectedRadiusY const): Deleted.
(WebCore::CSSEllipseValue::protectedCenterX const): Deleted.
(WebCore::CSSEllipseValue::protectedCenterY const): Deleted.
(WebCore::CSSPolygonValue::windRule const): Deleted.
(WebCore::CSSRectShapeValue::top const): Deleted.
(WebCore::CSSRectShapeValue::right const): Deleted.
(WebCore::CSSRectShapeValue::bottom const): Deleted.
(WebCore::CSSRectShapeValue::left const): Deleted.
(WebCore::CSSRectShapeValue::protectedTop const): Deleted.
(WebCore::CSSRectShapeValue::protectedRight const): Deleted.
(WebCore::CSSRectShapeValue::protectedBottom const): Deleted.
(WebCore::CSSRectShapeValue::protectedLeft const): Deleted.
(WebCore::CSSRectShapeValue::topLeftRadius const): Deleted.
(WebCore::CSSRectShapeValue::topRightRadius const): Deleted.
(WebCore::CSSRectShapeValue::bottomRightRadius const): Deleted.
(WebCore::CSSRectShapeValue::bottomLeftRadius const): Deleted.
(WebCore::CSSRectShapeValue::protectedTopLeftRadius const): Deleted.
(WebCore::CSSRectShapeValue::protectedTopRightRadius const): Deleted.
(WebCore::CSSRectShapeValue::protectedBottomRightRadius const): Deleted.
(WebCore::CSSRectShapeValue::protectedBottomLeftRadius const): Deleted.
(WebCore::CSSXywhValue::insetX const): Deleted.
(WebCore::CSSXywhValue::insetY const): Deleted.
(WebCore::CSSXywhValue::width const): Deleted.
(WebCore::CSSXywhValue::height const): Deleted.
(WebCore::CSSXywhValue::protectedInsetX const): Deleted.
(WebCore::CSSXywhValue::protectedInsetY const): Deleted.
(WebCore::CSSXywhValue::protectedWidth const): Deleted.
(WebCore::CSSXywhValue::protectedHeight const): Deleted.
(WebCore::CSSXywhValue::topLeftRadius const): Deleted.
(WebCore::CSSXywhValue::topRightRadius const): Deleted.
(WebCore::CSSXywhValue::bottomRightRadius const): Deleted.
(WebCore::CSSXywhValue::bottomLeftRadius const): Deleted.
(WebCore::CSSXywhValue::protectedTopLeftRadius const): Deleted.
(WebCore::CSSXywhValue::protectedTopRightRadius const): Deleted.
(WebCore::CSSXywhValue::protectedBottomRightRadius const): Deleted.
(WebCore::CSSXywhValue::protectedBottomLeftRadius const): Deleted.
(WebCore::CSSPathValue::pathData const): Deleted.
(WebCore::CSSPathValue::windRule const): Deleted.
* Source/WebCore/css/CSSCounterValue.h:
(WebCore::CSSCounterValue::identifier const): Deleted.
(WebCore::CSSCounterValue::separator const): Deleted.
(WebCore::CSSCounterValue::counterStyle const): Deleted.
* Source/WebCore/css/CSSCrossfadeValue.h:
* Source/WebCore/css/CSSCursorImageValue.h:
* Source/WebCore/css/CSSCustomPropertyValue.h:
* Source/WebCore/css/CSSFilterImageValue.h:
* Source/WebCore/css/CSSFontFaceSrcValue.cpp:
(WebCore::CSSFontFaceSrcResourceValue::customMayDependOnBaseURL const):
* Source/WebCore/css/CSSFontFaceSrcValue.h:
* Source/WebCore/css/CSSFontPaletteValuesOverrideColorsValue.h:
* Source/WebCore/css/CSSFontStyleRangeValue.h:
* Source/WebCore/css/CSSFontStyleWithAngleValue.h:
* Source/WebCore/css/CSSFontValue.cpp:
(WebCore::CSSFontValue::customVisitChildren const):
* Source/WebCore/css/CSSFontValue.h:
* Source/WebCore/css/CSSGradientValue.h:
* Source/WebCore/css/CSSImageSetOptionValue.h:
* Source/WebCore/css/CSSImageValue.cpp:
(WebCore::CSSImageValue::customMayDependOnBaseURL const):
* Source/WebCore/css/CSSImageValue.h:
* Source/WebCore/css/CSSOffsetRotateValue.h:
* Source/WebCore/css/CSSPendingSubstitutionValue.h:
(WebCore::CSSPendingSubstitutionValue::customVisitChildren const):
* Source/WebCore/css/CSSPrimitiveValue.cpp:
(WebCore::CSSPrimitiveValue::customVisitChildren const):
* Source/WebCore/css/CSSPrimitiveValue.h:
* Source/WebCore/css/CSSRayValue.h:
* Source/WebCore/css/CSSReflectValue.h:
* Source/WebCore/css/CSSScrollValue.h:
* Source/WebCore/css/CSSShadowValue.h:
* Source/WebCore/css/CSSValue.cpp:
(WebCore::CSSValue::visitChildren const):
(WebCore::CSSValue::mayDependOnBaseURL const):
* Source/WebCore/css/CSSValue.h:
(WebCore::CSSValue::customMayDependOnBaseURL const):
(WebCore::CSSValue::customVisitChildren const):
* Source/WebCore/css/CSSValueList.h:
(WebCore::CSSValueContainingVector::customVisitChildren const):
* Source/WebCore/css/CSSValuePair.h:
(WebCore::CSSValuePair::customVisitChildren const):
* Source/WebCore/css/CSSVariableReferenceValue.h:
(WebCore::CSSVariableReferenceValue::customMayDependOnBaseURL const):
* Source/WebCore/css/CSSViewValue.h:
* Source/WebCore/css/StyleProperties.cpp:
(WebCore::StyleProperties::mayDependOnBaseURL const):
* Source/WebCore/css/StyleProperties.h:
* Source/WebCore/css/StyleSheetContents.cpp:
(WebCore::StyleSheetContents::StyleSheetContents):
(WebCore::StyleSheetContents::isCacheable const):
(WebCore::StyleSheetContents::isCacheableAndNoBaseURLDependency const):
(WebCore::StyleSheetContents::mayDependOnBaseURL const):
* Source/WebCore/css/StyleSheetContents.h:
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::mayDependOnBaseURL):
* Source/WebCore/css/parser/CSSParserContext.h:
* Source/WebCore/dom/InlineStyleSheetOwner.cpp:
(WebCore::InlineStyleSheetOwner::createSheet):
(WebCore::makeStyleSheetContentsCacheKey): Deleted.

Canonical link: https://commits.webkit.org/275795@main
